apps/prairielearn/src/components/DeleteCourseInstanceModal.tsx

```diff
@@ -0,0 +1,194 @@
+                    I understand that <strong>{enrolledCount}</strong> enrolled{' '}
+                    {enrolledCount === 1 ? 'student' : 'students'} will lose access to this course
+                    instance.
+                  </label>
+                </div>
+                <div class="form-check mb-3">
+                  <input
+                    class="form-check-input"
+                    type="checkbox"
+                    id="irreversible"
+                    checked={checks.irreversible}
+                    onChange={(e) =>
+                      setChecks((c) => ({ ...c, irreversible: e.currentTarget.checked }))
+                    }
+                  />
+                  <label class="form-check-label" for="irreversible">
+                    I understand that once deleted, this course instance cannot be restored through
+                    the PrairieLearn web interface.
+                  </label>
+                </div>
+                <div class="form-check mb-2">
+                  <input
+                    class="form-check-input"
+                    type="checkbox"
+                    id="recovery"
+                    checked={checks.recovery}
+                    onChange={(e) =>
+                      setChecks((c) => ({ ...c, recovery: e.currentTarget.checked }))
+                    }
+                  />
+                  <label class="form-check-label" for="recovery">
+                    I understand data recovery requires manual Git operations on the course
+                    repository.
+                  </label>
+                </div>
+              </div>
+            )}
+            {step === 2 && (
+              <div>
+                <div class="card text-bg-light mb-3">
+                  <div class="card-body">
+                    <h3 class="h5 card-title">Recovery information</h3>
+                    <p>If you need to recover this course instance, you will need to:</p>
+                    <ol class="mb-0">
+                      <li>Access the course's Git repository</li>
+                      <li>
+                        Use Git commands to restore the course instance from the commit history
+                      </li>
+                      <li>Push your changes</li>
+                      <li>Sync your changes to PrairieLearn</li>
+                    </ol>
+                  </div>
+                </div>
+                <div>
+                  <label htmlFor="confirmShortName" class="form-label">
+                    Type <strong>{shortName}</strong> to confirm deletion:
+                  </label>
+                  <input
+                    ref={confirmationTextRef}
+                    id="confirmShortName"
+                    name="confirmShortName"
+                    class="form-control"
+                    value={confirmationText}
+                    autoComplete="off"
+                    onInput={(e) => setConfirmationText(e.currentTarget.value)}
```

Comment:
nit: I have been using `onChange`, not sure if there is any meaningful difference, but consistency one way or the other would be nice.

Reply:
Relevant:

- https://github.com/facebook/react/issues/9567
- https://preactjs.com/guide/v10/differences-to-react/#use-oninput-instead-of-onchange

Using `onChange` here would break the form; try it for yourself!

Reply:
Should we flag that as a lint violation? TIL that they are equivalent.

Reply:
`onChange` may still for checkboxes/radios/selects, I'm hesitant to ban it outright. We could try banning it just for `<input>`?